docs/intro/intro_symbolic_math.rst

```diff
@@ -4,96 +4,187 @@
 Symbolic Math
 =============
 
-dimod enables easy incorporation of binary and integer variables as
-:ref:`single-variable models <generators_symbolic_math>`. For example, you can
-represent such binary variables as follows:
+*dimod*'s support for symbolic math can simplify your coding of problems. For
+example, consider a problem of finding the rectangle with the greatest area for
+a given perimeter, which you can formulate mathematically as,
 
->>> from dimod import Binary, Spin
->>> x = Binary('x')
->>> s = Spin('s')
->>> x
-BinaryQuadraticModel({'x': 1.0}, {}, 0.0, 'BINARY')
+.. math::
 
-Similarly for integers:
+  \max_{i,j} \quad ij
 
->>> from dimod import Integer
->>> i = Integer('i')
+  \textrm{s.t.} \quad 2i+2j \le P
+
+where the components are,
+
+* **Variables**: :math:`i` and :math:`j` are the lengths of two sides of the
+  rectangle and and :math:`P` is the length of the perimeter.
+* **Objective**: maximize the area, which is given by the standard geometric
+  formula :math:`ij`.
+* **Constraint**: subject to (s.t.) not exceeding the given perimeter length;
+  that is, :math:`2i+2j`, the summation of the rectangle's four sides, is
+  constrained to a maximum value of :math:`P`.
+
+*dimod*'s symbolic math enables an intuitive coding of such problems:
+
+>>> print(objective.to_polystring())            # doctest:+SKIP
```

Comment:
I had to read this a bunch of times, but I think the first code snippet should be standalone. My first thought is "what is objective?" Something like
```python
>>> i, j = dimod.Binaries(['i', 'j'])
>>> objective = -i*j
>>> constraint = 2*i + 2*j <= 8
```
I think conveys the same point, while also being working code.

On that note, I would adjust the math above to include the word "objective" and "constraint", something like
```tex
\textrm{objective: } \max_{i,j} \quad ij
\textrm{constraint: } \textrm{s.t.} \quad 2i+2j \le 8
```
^ ideally with the alignment fixed, but you get the idea.

Note that I've also used 8 directly. IMO that makes the link super clear between code and math.

Reply:
> ```python
> >>> i, j = dimod.Binaries(['i', 'j'])
> ```

Just I used ``i, j = dimod.Integers(["i", "j"])``